lib/CLI/Types.hs

```diff
@@ -327,6 +324,102 @@ instance FromHttpApiData (FieldName, FilterField) where
               return (field, FilterFieldByDate op date)
           ]
 
+-- These @ToJSON@ instances are for success results only.
```

Comment:
If error produces empty object json, do we really need `"result": "ok"` at all?

Reply:
At this moment all errors are produced in `Server` module. It's expected that if we call `toJSON` on results, the result is successful. I made the "result" field to understand whether our action was successful or not.

Reply:
Oh, I see. But isn't responding with error code enough (which you already do) to understand that operation was unsuccessful? (I have little experience in web programming so my opinion can be groundless haha 😂)

Reply:
>  But isn't responding with error code enough (which you already do) to understand that operation was unsuccessful?

Yeah, I agree. A consumer of our API should look at the status code - if it's a 2xx, they should decode the response body to a directory/entry/whatever, and if it's any other status code they should expect the response body to have the format `{ "error": "", "code": n }`

---

A separate, but related issue, is that we now have instances like this:

```hs
instance ToJSON CreateResult where
  toJSON = \case
    CRSuccess entryPath ->
      [aesonQQ|
        {
          "contents": #{entryPath},
          "result": "ok"
        }
      |]
    _ -> emptyObject
```

Where `toJSON` is safe to use with _some values_ (`CRSuccess`), but when used with other values (`CRCreateError`) it'll produce meaningless output, which kinda goes ahead haskell's mantra.

Reply:
I don't know how to write instances for errors. I depend on functions from `Coffer.PrettyPrint` but I can't import them because of cyclic dependencies.